lib/Parse/ParseDecl.cpp

```diff
@@ -7416,6 +7416,12 @@ Parser::parseDeclSubscript(SourceLoc StaticLoc,
 
   diagnoseWhereClauseInGenericParamList(GenericParams);
 
+  // Protocol requirement arguments may not have default values.
+  if (Flags.contains(PD_InProtocol) && DefaultArgs.HasDefaultArgument) {
```

Comment:
Longer term, it would be better to diagnose as much as possible in the type checker and not the parser, in my opinion. This could be checked in TypeCheckDeclPrimary.cpp, along with other related prohibitions on default arguments in protocol members.

Reply:
Right, this should be a temporary fix to align `subscript` with `func`/`init`. We should move those diagnostics to type checker later.